### DISCUSSION TITLE
Code-only Format

### DISCUSSION THREAD

**Discussion by jdc-pub** [quarto-dev/quarto-cli#3625] [Q&A]
Is there any supported way to output only code, like a `script` format?

Something like...

```
---
title: Code Only
format:
  script: default
---

# Section \#1

'''{python}
def square(x): return x**2
'''

```

Would produce `Code\ Only.py`.

**Comment by baptiste** [comment]
The closest I can think of is `knitr::purl()`, but it's kinda R-only (the python chunks seem to get commented out). Might be a useful starting point for a custom format though.

````
```{r, echo=FALSE, results='hide'}
knitr::purl(input = knitr::current_input(), output = glue::glue('{rmarkdown::metadata$title}.py'))
```
````

**Reply by baptiste** [reply]
A more python-friendly solution might be to first convert the qmd document into a notebook, via

> `quarto convert`

followed by

> `jupyter nbconvert --to script`

to convert it into a script. But I agree it'd be nice to have a built-in `quarto purl` command to do it in a language-agnostic way (maybe producing separate files if there is more than one language in the source). To me this would feel more natural than implemented as an output "format".
